isb_lib/opencontext_adapter.py

```diff
@@ -0,0 +1,175 @@
+        _page_size = self._page_size
+        params = {
+            "rows": _page_size,
+        }
+        more_work = True
+        num_records = 0
+        while more_work:
+            L.info("trying to hit %s", self.url)
+            response = requests.get(
+                self.url, params=params, headers=headers, timeout=HTTP_TIMEOUT
+            )
+
+            if response.status_code != 200:
+                L.error(
+                    "Unable to load records; status: %s; reason: %s",
+                    response.status_code,
+                    response.reason,
+                )
+                break
+            # L.debug("recordsInProject data: %s", response.text[:256])
+            data = response.json()
+            for record in data.get("oc-api:has-results", {}):
+                L.info("records_in_page Record id: %s", record.get("uri", None))
+                # print(json.dumps(record, indent=2))
+                # raise NotImplementedError
+                yield record
+                num_records += 1
+            self.url = data["next-json"]
+            if len(data.get("oc-api:has-results", {})) < _page_size:
+                more_work = False
+            elif 0 < self._max_entries <= num_records:
+                more_work = False
+            elif num_records == self._page_size:
+                more_work = False
+
+    def loadEntries(self):
+        self._cpage = []
+        self._page_offset = 0
+        counter = 0
+        for item in self.records_in_page():
+            self._cpage.append(item)
+            counter += 1
+            if 0 < self._max_entries < counter:
+                break
+        self._total_records = len(self._cpage)
+
+    def _getPage(self):
+        """Override this method to retrieve a page of entries from the service.
+
+        After completion, self._cpage contains the next page of entries or None if there
+        are no more pages available, and self._page_offset is set to the first entry (usually 0)
+
+        """
+        is_none = self._cpage is None
+        if is_none or self._page_offset >= len(self._cpage):
+            if not is_none:
+                # Work around the rate limiter
+                time.sleep(0.3)
+            self.loadEntries()
+        if self._coffset >= self._total_records:
+            return
+        self._page_offset = 0
+
+    def last_url_str(self) -> typing.AnyStr:
+        return self.url
+
+
+def load_thing(
+    thing_dict: typing.Dict, t_resolved: datetime.datetime, url: typing.AnyStr
+) -> igsn_lib.models.thing.Thing:
+    """
+    Load a thing from its source.
+
+    Minimal parsing of the thing is performed to populate the database record.
+
+    Args:
+        thing_dict: Dictionary representing the thing
+        t_resolved: When the item was resolved from the source
+
+    Returns:
+        Instance of Thing
+    """
+    L = get_logger()
+    id = thing_dict["uri"]
+    t_created = dateparser.parse(thing_dict.get("published"))
```

Comment:
Another difference here -- OpenContext records don't have a creation date the way that GEOME does, so we use the published date as a best placeholder.